### ISSUE TITLE
Not able to copy the GetObjectResponse.ResponseStream to another stream

### ISSUE THREAD

**Issue by bli88** [aws/aws-sdk-net#1163]
When trying to copy the ResponseStream for GetObjectAsync call to a separate stream, nothing get copied. The code example is like following:
GetObjectResponse response = null;
try
{
        response = await _amazonS3Client.GetObjectAsync(bucket, key);
}
catch (AmazonS3Exception amazonS3Exception)
{
.....
}
MemoryStream stream = new MemoryStream();
await response.ResponseStream.CopyToAsync(stream);

## Expected Behavior
The returned Stream can be copied to another stream for further processing. 

## Current Behavior
Zero bytes of the content from the returned stream will be copied. 

## Your Environment
* AWSSDK.Core version used: 3.3.30.2
* Service assembly and version used: S3 3.3.31.5
* Operating System and version: Windows10
* Visual Studio version: 2017
* Targeted .NET platform: .NET Core

## .NET Core Info

* .NET Core version used for development: 2.1
* .NET Core version installed in the environment where application runs: 2.1
* Output of `dotnet --info`: 
.NET Core SDK (reflecting any global.json):
 Version:   2.1.403
 Commit:    04e15494b6

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.16299
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.403\

Host (useful for support):
  Version: 2.1.5
  Commit:  290303f510

.NET Core SDKs installed:
  1.1.10 [C:\Program Files\dotnet\sdk]
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.403 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
* Contents of project.json/project.csproj: 

[logging]: http://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-other.html

**Comment by klaytaybai** [comment]
Hi @bli88, thanks for the additional debugging info.  I suspected that might be the issue from your first comment.  We'll investigate further.

What is going on in your catch?  Are you somehow setting the response?  It looks like your code will throw a null reference

**Comment by bli88** [comment]
Hi @klaytaybai,
Thanks for the response. With the additional information capture, I was able to determine the root cause of the problem. Like you've suspected, it's related to the CurrentPosition in the stream. The capture was after I've written the file to a local device and that's why the CurrentPosition became the last position. Since the stream is with property of CanSeek with false, I cannot reset the position to 0 for re-reading. 
After I removed the output to the local device, I was able to read the content into another stream.

Thanks again for your support. I'll close the issue.
